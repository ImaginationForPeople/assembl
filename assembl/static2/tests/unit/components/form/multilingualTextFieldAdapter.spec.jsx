import React from 'react';
import renderer from 'react-test-renderer';

import DummyForm from './dummyForm';
import MultilingualTextFieldAdapter from '../../../../js/app/components/form/multilingualTextFieldAdapter';

describe('MultilingualTextFieldAdapter component', () => {
  const onChangeSpy = jest.fn();
  const onFocusSpy = jest.fn();

  it('should render a text field', () => {
    const props = {
      editLocale: 'fr',
      input: {
        name: 'title',
        onChange: onChangeSpy,
        onFocus: onFocusSpy,
        value: {
          en: 'Hello',
          fr: 'Bonjour'
        }
      },
      label: 'Title',
      meta: {
        error: '',
        touched: false
      }
    };
    const component = renderer.create(
      <DummyForm>
        <MultilingualTextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a required text field', () => {
    const props = {
      editLocale: 'fr',
      input: {
        name: 'title',
        onChange: onChangeSpy,
        onFocus: onFocusSpy,
        value: {
          en: 'Hello',
          fr: 'Bonjour'
        }
      },
      label: 'Title',
      meta: {
        error: '',
        touched: false
      },
      required: true
    };
    const component = renderer.create(
      <DummyForm>
        <MultilingualTextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text field without value', () => {
    const props = {
      editLocale: 'fr',
      input: {
        name: 'title',
        onChange: onChangeSpy,
        onFocus: onFocusSpy
      },
      label: 'Title',
      meta: {
        error: '',
        touched: false
      }
    };
    const component = renderer.create(
      <DummyForm>
        <MultilingualTextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text field untouched with error', () => {
    const props = {
      editLocale: 'fr',
      input: {
        name: 'title',
        onChange: onChangeSpy,
        onFocus: onFocusSpy
      },
      label: 'Title',
      meta: {
        error: 'This field is required',
        touched: false
      }
    };
    const component = renderer.create(
      <DummyForm>
        <MultilingualTextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render a text field touched with error', () => {
    const props = {
      editLocale: 'fr',
      input: {
        name: 'title',
        onChange: onChangeSpy,
        onFocus: onFocusSpy
      },
      label: 'Title',
      meta: {
        error: 'This field is required',
        touched: true
      }
    };
    const component = renderer.create(
      <DummyForm>
        <MultilingualTextFieldAdapter {...props} />
      </DummyForm>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});