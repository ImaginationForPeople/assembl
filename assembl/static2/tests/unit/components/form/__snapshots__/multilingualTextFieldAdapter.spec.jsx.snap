// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultilingualTextFieldAdapter component should render a required text field 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="title"
    >
      Title *
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Title *"
      required={true}
      type={undefined}
      value="Bonjour"
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`MultilingualTextFieldAdapter component should render a text field 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <label
      className="control-label"
      htmlFor="title"
    >
      Title
    </label>
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Title"
      required={false}
      type={undefined}
      value="Bonjour"
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`MultilingualTextFieldAdapter component should render a text field touched with error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group has-error"
  >
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Title"
      required={false}
      type={undefined}
      value=""
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`MultilingualTextFieldAdapter component should render a text field untouched with error 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Title"
      required={false}
      type={undefined}
      value=""
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;

exports[`MultilingualTextFieldAdapter component should render a text field without value 1`] = `
<form
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="title"
      onChange={[Function]}
      onFocus={[Function]}
      placeholder="Title"
      required={false}
      type={undefined}
      value=""
    />
  </div>
  <button
    disabled={true}
    type="submit"
  >
    Submit
  </button>
</form>
`;
